src/components/Tasks/TaskList.tsx

```diff
@@ -104,30 +108,71 @@ interface PropsType {
   collections: EteSync.CollectionInfo[];
   onItemClick: (entry: TaskType) => void;
   onItemSave: (item: PimType, journalUid: string, originalItem?: PimType) => Promise<void>;
+  syncInfo: SyncInfo;
+  userInfo: UserInfoData;
+  etesync: CredentialsData;
 }
 
 export default function TaskList(props: PropsType) {
   const [showCompleted, setShowCompleted] = React.useState(false);
   const [showHidden, setShowHidden] = React.useState(false);
   const [searchTerm, setSearchTerm] = React.useState('');
-  const [info, setInfo] = React.useState('');
+  const [toast, setToast] = React.useState<{ message: string, severity: ToastProps['severity'] }>({ message: '', severity: undefined });
```

Comment:
How is this change related to this commit? I'll let it slide this time, but it feels like quite the unrelated change, which made this review harder than it should have been.

Reply:
I felt it was related because I wanted to have error handling in the new sync code. It seemed weird to be using `setInfo` for an error, so I expanded it. I can split up the changes if you'd like, but the code is fairly coupled.

Reply:
No worries, it was just a bit confusing.

Reply:
P.S, I make some nit-picky comments every now and then, and I make the same comments to myself too. :P 
What I'm trying to say: don't worry about them too much. :)

Reply:
Yeah I sometimes struggle with where to divide my commits, especially when developing things where two features are distinct yet rely on each other. It's like I don't want to push an incomplete feature, but I also don't want the changes to be too broad.